### PR TITLE
Feature/service refactor

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Integration of pf::services::manager framework into pfcmd and the webgui.

Team I created a new branch feature/service-refactor for the new and improved pfcmd service command.

Notable changes 
- each service has it's own service manager perl module
- pfcmd pf watch - Will now only restart service that are down
- pfcmd <service> stop|start will show if the service was already stopped or started
- pfcmd service is now in color!

Questions 
- Should we have a configuration setting that force an entire restart of pf if any service is down when the watch is done.
  servicewatch.restart_type=soft|hard
- Is there anything else we should add since we are in the code at the moment?

### CODE REVIEW

File: lib/pf/services/manager/memcached.pm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+use warnings;
+use Moo;
+use pf::file_paths;
+
+extends 'pf::services::manager';
+with 'pf::services::manager::roles::pf_conf_service_managed';
+
+has '+name' => (default => sub { 'memcached' } );
+
+has '+launcher' => (default => sub { "%1\$s -d -p 11211 -u pf -m 64 -c 1024 -P $install_dir/var/run/memcached.pid"});
```
Comment (dwlfrth) [inline lib/pf/services/manager/memcached.pm:24]: Comments...
"Random" values are useless even if needed when undocumented

**Comment by dwlfrth** [comment]
Can you document (pf::services::manager will probably be the best place) all possibles method (mainly for overloading purposes but also to have a good idea of what which method is doing) and the startup/stop flow.

**Comment by dwlfrth** [comment]
What is the purpose of pf::services::manager::roles::pf_conf_service_managed ?

**Comment by jrouzierinverse** [comment]
pf::services:: manager is pretty much documented.

**Pull request merged**